### DISCUSSION TITLE
Login error

### DISCUSSION THREAD

**Discussion by Muhammad0320** [idurar/idurar-erp-crm#285] [Q&A]
I want to login with the provided credentials in the readme.md file it greeted me with errors like no email with this was registered

**Comment by salahlalami** [comment]
could you please provide us with screenshoot of network response

**Comment by MSNoorul** [comment]
![image](https://github.com/idurar/idurar-erp-crm/assets/111111285/b97b44b2-e9d0-4b9f-ad86-fc3070bc8248)
i am interested to work on opensource i run this on local host but how to i login there is no register page only log in page

**Reply by bruno-keiko** [reply]
Hello. Did you solve this problem?

**Reply by MSNoorul** [reply]
yes see my githup readme file i add a section to how to login then update also give pull request to merge

**Reply by NandaniThakur** [reply]
Was there there any issue raised for this PR which you have created ?
from where you have got those username and password credentials?

**Reply by AhsanSarwar0413** [reply]
> yes see my githup readme file i add a section to how to login then update also give pull request to merge

you are lying.

**Comment by bruno-keiko** [comment]
I have the same issue, I am unable to log in. Who has accomplished this problem please help me! Thank you!

**Comment by NandaniThakur** [comment]
if someone have found /will find then please update here.

**Comment by MSNoorul** [comment]
username : admin@demo.com - password : admin123 
use this credentials

**Comment by MSNoorul** [comment]
also check my forked repo i add clear description to how to login in me readme doc

**Comment by NandaniThakur** [comment]
I was unable to find description in readme.md file. If you can please provide me link.
 also I have used above login credentials but nothing worked.

**Comment by Slacky300** [comment]
### Setting Up the Backend for Local Login to idurar-erp-crm

1. **Update the `.env` file in the backend folder:**

   Open the `.env` file in the backend folder and update the `DATABASE` variable with your MongoDB URI:

   ```bash
   DATABASE="your-mongodb-uri"
   ```
   - If you have MongoDB Compass installed locally, you can use:
   
   ```bash
     DATABASE="mongodb://localhost:27017/db-name"
   ```
   - Alternatively, you can use your MongoDB Atlas URL.

2. Install backend dependencies and run the setup command:

    - After saving the .env file, make sure you are in the backend directory, then install the necessary dependencies and run the setup 
    command:

 
    ```bash

    npm install
    npm run setup
    ```

   - Screenshot of the code of setup.js
  
    ![setup.js](https://github.com/user-attachments/assets/0caca32e-d3fb-4489-8fba-6adce95fc60c)

3. This will create the necessary schemas and a default admin account with the credentials:

    - Email: admin@demo.com
    - Password: admin123

**Comment by JivithJJ** [comment]
how to generate the jwt secret token?, im unable to login with the creds given above. do share your suggestions

**Reply by Slacky300** [reply]
> how to generate the jwt secret token?, im unable to login with the creds given above. do share your suggestions

You can use any string as the JWT secret, but it's recommended to generate a secure one using a website like [jwt.io](https://jwt.io/) or by running the following command in your terminal:  

```sh
node -e "console.log(require('crypto').randomBytes(32).toString('hex'))"
```

Once generated, paste it into your `backend/.env` file as:  

```env
JWT_SECRET=your_generated_secret
```

After doing this, restart your backend server and try logging in again.
